### PR TITLE
add randutil, use fastrand as random source

### LINKED ISSUE #3537: Use of weak random number generator (math/rand instead of crypto/rand)
### Description of the code refactor
fix all [Use of weak random number generator (math/rand instead of crypto/rand)](https://github.com/iotexproject/iotex-core/security/code-scanning) in security scan result (page 2 & 3)

### Why is this needed
- [] better/less memory usage
- [] reduce running time
- [] clean (easy-to-understand) data structure or algorithm
- [] remove dead code
- [x] better code design or structure (potential security issue)
- [] reduce/decouple dependencies between code modules
- [] increase test coverage
- [] better code readability

### PR DESCRIPTION
**Pull request by millken**
# Description
fix code scanning error G404

Fixes #3537 

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/util/randutil/randutil.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package randutil
```
Comment (Liuhaai) [inline pkg/util/randutil/randutil.go:1]: why is this added

Comment (millken) [reply pkg/util/randutil/randutil.go:1]: Only `math/rand` have func `Intn`, need to implement this common func yourself

Comment (Liuhaai) [reply pkg/util/randutil/randutil.go:1]: https://pkg.go.dev/math/rand

**Comment by CoderZhi** [comment]
all the random numbers used in our package so far has no impact to product, especially those used in unit tests. Thus, this PR may be an overreact to CodeQL, which we should prevent.

**Comment by dustinxie** [comment]
closed per @CoderZhi  comment, these random number only used in test, does not affect randomness in our consensus or block building/signing process
